File: 1768-merge-strings-alternately/1768-merge-strings-alternately.js
/**
 * @param {string} word1
 * @param {string} word2
 * @return {string}
 */
var mergeAlternately = function(word1, word2) {
    let i = 0;
    let j = 0;
    let common_string = ""
    while (i < word1.length && j < word2.length){
        common_string += word1[i]
        common_string += word2[j]
        i += 1
        j += 1
    }
    
    while (i < word1.length){
        common_string += word1[i]
        i += 1
    }
    
    while (j < word2.length){
        common_string += word2[j]
        j += 1
    }
    
    return common_string
};

File: 1768-merge-strings-alternately/1768-merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        
        final_string = ""
        
        i, j = 0, 0
        while i < len(word1) and j < len(word2):
            final_string += word1[i]
            final_string += word2[j]
            i += 1
            j += 1
            
        while i < len(word1):
            final_string += word1[i]
            i += 1
            
        while j < len(word2):
            final_string += word2[j]
            j += 1
            
        return final_string
        